### PR TITLE
Fix missing credentials when fetching tarballs

### PR DESCRIPTION
**Pull request by rtsao**
**What's the problem this PR addresses?**

Fixes https://github.com/yarnpkg/berry/issues/601

**How did you fix it?**

- Use `npmHttpUtils.get` in `NpmHttpFetcher`
- Add support for absolute paths in `npmHttpUtils.get`

### CODE REVIEW

**Comment by arcanis** [comment]
I'm not sure about this fix - in particular the `%2f` codepath should probably be kept exclusive to the semver fetcher, in which case there's little point factorizing this code.

Imo, the best fix would be to simply make `NpmHttpFetcher` use `npmHttpUtils.get` instead of `httpUtils.get`.

**Comment by rtsao** [comment]
> I'm not sure about this fix - in particular the `%2f` codepath should probably be kept exclusive to the semver fetcher, in which case there's little point factorizing this code.
> 
> Imo, the best fix would be to simply make `NpmHttpFetcher` use `npmHttpUtils.get` instead of `httpUtils.get`.

Got it, will update accordingly.

What do you think would be the ideal way to test this?

File: packages/plugin-npm/sources/NpmHttpFetcher.ts [hunk 1]
```diff
@@ -50,13 +51,9 @@ export class NpmHttpFetcher implements Fetcher {
     if (archiveUrl === null)
       throw new Error(`Assertion failed: The archiveUrl querystring parameter should have been available`);
 
-    const sourceBuffer = await httpUtils.get(archiveUrl, {
+    return npmHttpUtils.get(archiveUrl, {
```
Comment (rtsao) [inline packages/plugin-npm/sources/NpmHttpFetcher.ts]: I'm not sure if stylistically you would prefer:

```suggestion
    return await npmHttpUtils.get(archiveUrl, {
```

File: packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts [hunk 2]
```diff
@@ -233,7 +233,7 @@ export const startPackageServer = (): Promise<string> => {
                 [version as string]: Object.assign({}, packageVersionEntry!.packageJson, {
                   dist: {
                     shasum: await getPackageArchiveHash(name, version),
-                    tarball: localName === `unconventional-tarball`
+                    tarball: (localName === `unconventional-tarball` || localName === `private-unconventional-tarball`)
```
Comment (rtsao) [inline packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts]: I was considering something like this (along the same lines of ``localName.startsWith(`private`)``):
```suggestion
                    tarball: (localName.includes(`unconventional-tarball`))
```

Comment (arcanis) [reply packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts]: It's ok for now - if we need to add more private packages we will just move the whitelist into a set 👍

**Comment by arcanis** [comment]
Awesome, thanks a lot @rtsao! 🎉

**Pull request merged**